### PR TITLE
[Python] fixed bug in api_client.py `__call_api` method

### PR DESCRIPTION
**Pull request by zorrofox**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR
fix the issue #6029

### CODE REVIEW

**Comment by wing328** [comment]
@zorrofox thanks for the PR. The fix looks good to me but I'll defer to the Python experts to have a look.

cc @frol @taxpon

**Comment by wing328** [comment]
cc @baartosz as well who performs the refactoring work.

**Comment by frol** [comment]
It seems to me that the target PR branch should be 2.3 instead of master.

**Comment by frol** [comment]
I think you can simply drop the `config` assignment and simply use `self.configuration.` directly. BTW, there is only a single place where it is used: https://github.com/swagger-api/swagger-codegen/blob/e1e5ac4d379234ac8d4525deb8fa7be983cc1e8c/modules/swagger-codegen/src/main/resources/python/api_client.mustache#L111

**Comment by wing328** [comment]
> It seems to me that the target PR branch should be 2.3 instead of master.

We released 2.2.3 (previous master) over the weekend and master is now 2.3.0.

File: samples/client/petstore/python/.swagger-codegen/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.2.3-SNAPSHOT
\ No newline at end of file
+2.3.0-SNAPSHOT
```
Comment (frol) [inline samples/client/petstore/python/.swagger-codegen/VERSION]: > We released 2.2.3 (previous master) over the weekend and master is now 2.3.0.

If this is the case, why does this diff and many other unrelated changes appear?

Anyway, the fix by itself is correct, so if the other matters are fine, it is good to merge this in.

Comment (wing328) [inline samples/client/petstore/python/.swagger-codegen/VERSION]: @frol thanks for reviewing. Very good questions. I'll update the samples after the merge (some changes are due to changes in Swagger Parser)
